File: Mathematical Algorithms/Sphenic Number/Sphenic.java
import java.util.Arrays;

public class Sphenic {
    static boolean[] arr = new boolean[1001];

    static void simpleSieve() {
        // initialize all entries of it as true. A value
        // in mark[p] will finally be false if 'p' is Not
        // a prime, else true.
        Arrays.fill(arr, true);

        // One by one traverse all numbers so that their
        // multiples can be marked as composite.
        for (int p = 2; p * p < 1001; p++) {

            // If p is not changed, then it is a prime
            if (arr[p]) {

                // Update all multiples of p
                for (int i = p * 2; i < 1001; i = i + p)
                    arr[i] = false;
            }
        }
    }

    static int find_sphene(int N) {
        int[] arr1 = new int[8];   // to store the 8 divisors
        int count = 0;        // to count the number of divisor
        int j = 0;
        for (int i = 1; i <= N; i++) {
            if (N % i == 0 && count < 8) {
                count++;
                arr1[j++] = i;

            }
        }

        // finally check if there re 8 divisor and
        // all the numbers are distinct prime no return 1
        // else return 0);
        if (count == 8 && (arr[arr1[1]] && arr[arr1[2]] && arr[arr1[3]]))
            return 1;

        return 0;
    }
}

File: Mathematical Algorithms/Sphenic Number/Sphenic.py
arr = [True] * (1001)

def simpleSieve():
    # Initialize all entries of it as
    # True. A value in mark[p] will
    # finally be False if 'p' is Not
    # a prime, else True.
    k = 0

    # One by one traverse all numbers so
    # that their multiples can be marked
    # as composite.
    for p in range(2, 1001):
        if p * p > 1001:
            break

        # If p is not changed, then it is a prime
        if arr[p]:

            # Update all multiples of p
            for k in range(p, 1001, k + p):
                arr[k] = False


def find_sphene(N):
    # To store the 8 divisors
    arr1 = [0] * (8)

    # To count the number of divisor
    count = 0
    j = 0

    for i in range(1, N + 1):
        if N % i == 0 and count < 8:
            count += 1
            arr1[j] = i
            j += 1

    # Finally check if there re 8 divisor and
    # all the numbers are distinct prime no return 1
    # else return 0);
    if (count == 8 and (arr[arr1[1]] and
                        arr[arr1[2]] and arr[arr1[3]])):
        return 1;

    return 0;